internal/services/loganalytics/log_analytics_linked_storage_account_resource.go

```diff
@@ -41,14 +41,15 @@ func resourceLogAnalyticsLinkedStorageAccount() *pluginsdk.Resource {
 				Type:     pluginsdk.TypeString,
 				Required: true,
 				ForceNew: true,
+				// TODO 4.0: change ignoreCase from true to false and remove diff suppress to respect correct casing
 				ValidateFunc: validation.StringInSlice([]string{
-					strings.ToLower(string(linkedstorageaccounts.DataSourceTypeCustomLogs)),
-					strings.ToLower(string(linkedstorageaccounts.DataSourceTypeAzureWatson)),
-					strings.ToLower(string(linkedstorageaccounts.DataSourceTypeQuery)),
-					strings.ToLower(string(linkedstorageaccounts.DataSourceTypeAlerts)),
-					// Value removed from enum in 2020-08-01, but effectively still works
-					"ingestion",
-				}, false),
+					string(linkedstorageaccounts.DataSourceTypeAlerts),
+					string(linkedstorageaccounts.DataSourceTypeAzureWatson),
+					string(linkedstorageaccounts.DataSourceTypeCustomLogs),
+					string(linkedstorageaccounts.DataSourceTypeIngestion),
+					string(linkedstorageaccounts.DataSourceTypeQuery),
+				}, true),
+				DiffSuppressFunc: suppress.CaseDifference,
```

Comment:
can we flag these with 4.0 - we want all 4.0 changes flagged so 4.0 is just a matter of flipping the flag
```suggestion
				}, !features.FourPointOhBeta),
				DiffSuppressFunc: !features.FourPointOhBeta,
```

Reply:
We cannot use "DiffSuppressFunc: !features.FourPointOhBeta" since DiffSuppressFunc requires dynamic value. So I flag it via another way. Please take another look. Thanks.